### PR TITLE
Add log messages on pool setup fail

### PR DESCRIPTION
**Pull request by tasleson**
Addresses: https://github.com/stratis-storage/stratisd/issues/779

### CODE REVIEW

**Comment by agrover** [comment]
:+1:

File: src/engine/strat_engine/engine.rs [hunk 1]
```diff
@@ -170,40 +173,46 @@ impl Engine for StratEngine {
...
+                        //
+                        // We will also _not_ exit the daemon as we do in initialize as we were
+                        // previously up and running for some duration of time.
+                        if !self.pools.contains_name(&pool_name) {
+                            self.pools.insert(pool_name, pool_uuid, pool);
+                            Some(pool_uuid)
+                        } else {
+                            let dev_paths = devices
+                                .values()
+                                .map(|p| p.to_str().expect("PathBuf.to_str for device path fail"))
```
Comment (mulkieran) [inline src/engine/strat_engine/engine.rs]: The expect() message should explain why this action must logically succeed.

Comment (tasleson) [reply src/engine/strat_engine/engine.rs]: @mulkieran Please state what you would like this expect statement to be, thanks!

Comment (tasleson) [reply src/engine/strat_engine/engine.rs]: I've selected the same text Andy used in a previous commit that has the same expectation of success.

**Pull request merged**